File: app/src/main/java/com/app/cancellingretrofitrequests/java/model/SearchResultModel.java

package com.app.cancellingretrofitrequests.java.model;

import com.google.gson.annotations.SerializedName;

@SuppressWarnings("unused")
public class SearchResultModel {

    @SerializedName("data")
    private Data mData;
    @SerializedName("status")
    private String mStatus;

    public Data getData() {
        return mData;
    }

    public void setData(Data data) {
        mData = data;
    }

    public String getStatus() {
        return mStatus;
    }

    public void setStatus(String status) {
        mStatus = status;
    }

}

File: app/src/main/java/com/app/cancellingretrofitrequests/kotlin/model/SearchResultModel.kt
package com.app.cancellingretrofitrequests.kotlin.model

data class SearchResultModel(
        val `data`: Data?,
        val status: String?
) {
    data class Data(
            val count_in_response: Int?,
            val items: List<Item?>?,
            val q: String?,
            val spelling_alternatives: SpellingAlternatives?,
            val start: Int?,
            val total_count: Int?
    ) {
        data class Item(
                val checksum: Checksum?,
                val dataset_citation: String?,
                val dataset_id: String?,
                val dataset_name: String?,
                val dataset_persistent_id: String?,
                val file_content_type: String?,
                val file_id: String?,
                val file_type: String?,
                val md5: String?,
                val name: String?,
                val published_at: String?,
                val size_in_bytes: Int?,
                val type: String?,
                val url: String?
        ) {
            data class Checksum(
                val type: String?,
                val value: String?
            )
        }

        class SpellingAlternatives(
        )
    }
}